Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -464,11 +496,47 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
     if shouldSkip(node) {
       return .skipChildren
     }
-    if node.identifier.presence == .missing,
-      let unexpected = node.unexpectedBeforeIdentifier,
-      unexpected.first?.as(TokenSyntax.self)?.tokenKind == .pound
-    {
-      addDiagnostic(unexpected, UnknownDirectiveError(unexpected: unexpected), handledNodes: [unexpected.id, node.identifier.id])
+    if node.identifier.presence == .missing, let unexpected = node.unexpectedBeforeIdentifier {
+      if unexpected.first?.as(TokenSyntax.self)?.tokenKind == .pound {
+        addDiagnostic(unexpected, UnknownDirectiveError(unexpected: unexpected), handledNodes: [unexpected.id, node.identifier.id])
+      } else if let availability = unexpected.first?.as(AvailabilityConditionSyntax.self) {
+        if let prefixOperatorExpr = node.parent?.as(PrefixOperatorExprSyntax.self),
```

Comment:
Might be worth splitting this out into its own method

Reply:
I factored out the logic for `negatedAvailabilityKeyword`. With that I think the logic is pretty straightforward again.